kong/plugins/cors/handler.lua

```diff
@@ -52,8 +52,24 @@ local function normalize_origin(domain)
   }
 end
 
+local function add_vary_header(header_filter)
+  if kong.ctx.plugin.vary_added then
+    return
+  end
+  kong.ctx.plugin.vary_added = true
+  local add_header = kong.response.add_header
+  local set_header = kong.response.set_header
+
+  if header_filter then
+    local vary_header = kong.response.get_header("Vary")
+    if vary_header then
+      return add_header("vary", "Origin")
+    end
+  end
+  set_header("vary", "Origin")
```

Comment:
Hi, please change this section to:

```suggestion
  if header_filter then
    kong.response.add_header("vary", "Origin")
  else
    kong.response.set_header("vary", "Origin")
  end
```

Reply:
Hi @kikito 
I've changed these lines as requested and I've updated tests because `if` statement was necessary to deduplicate vary header.

But I've added one new fix(I think) to this PR.  Vary header is always returned when using cors plugin.
Why?
A very good explanation can be found here https://github.com/rs/cors/issues/10

I'm open to discussion and I'm waiting for feedback.